src/modules/roc_fec/target_openfec/roc_fec/of_decoder.cpp

```diff
@@ -71,6 +71,16 @@ bool OFDecoder::valid() const {
     return valid_;
 }
 
+size_t OFDecoder::max_block_length() const {
+    roc_panic_if_not(valid());
+
+    if (codec_id_ == OF_CODEC_REED_SOLOMON_GF_2_M_STABLE) {
+        return OF_REED_SOLOMON_MAX_NB_SOURCE_SYMBOLS_DEFAULT;
+    }
+
+    return OF_LDPC_STAIRCASE_MAX_NB_SOURCE_SYMBOLS_DEFAULT;
```

Comment:
I suggest a switch with a panic in default branch, so that we notice the problem if we add a new FEC scheme ad forget to patch this method.

Reply:
Done